File: UtilitiesVB/Interfaces/IProjectInfo.vb
Public Interface IProjectInfo
    Sub Save()
    Sub Save(FileName_IDList As String)
    Function Contains_ProgramName(StrProgramName As String) As Boolean
    Function Contains_ProjectID(StrProjectID As String) As Boolean
    Function Contains_ProjectName(StrProjectName As String) As Boolean
    Function Find_ByProgramName(StrProgramName As String) As List(Of IToDoProjectInfoEntry)
    Function Find_ByProjectID(StrProjectID As String) As List(Of IToDoProjectInfoEntry)
    Function Find_ByProjectName(StrProjectName As String) As List(Of IToDoProjectInfoEntry)
    Function Programs_ByProjectNames(StrProjectNames As String) As String
End Interface

File: UtilitiesCS/Interfaces/IToDo/IProjectInfo.cs
using System;
using System.Collections.Generic;


namespace UtilitiesCS
{
    public interface IProjectInfo: ISerializableList<IToDoProjectInfoEntry>
    {
        bool Contains_ProgramName(string programName);
        bool Contains_ProjectID(string projectID);
        bool Contains_ProjectName(string projectName);
        List<IToDoProjectInfoEntry> Find_ByProgramName(string programName);
        List<IToDoProjectInfoEntry> Find_ByProjectID(string projectID);
        List<IToDoProjectInfoEntry> Find_ByProjectName(string projectName);
        string Programs_ByProjectNames(string projectNames);
        void Save();
        void Save(string filepath);
        (bool Any, int[] Indices) IsCorrupt();
        void Rebuild(Microsoft.Office.Interop.Outlook.Application olApp);
        void SetIdUpdateAction(Action<string, string> action);
    }
}